### PR TITLE
fix(attached-android): reverse port forwarding, and screen unlocking

### PR DESCRIPTION
**Pull request by mnzaki**
- [x] This is a small change 

---

**Description:**
This PR addresses 2 things for testing on android devices:
- Automatic reverse port forwarding for `DetoxServer` to connect back to the host machine. Relevant issues: #1174 #1200 #968 
- Screen unlocking from `'Dozing'` state

### CODE REVIEW

**Comment by mnzaki** [comment]
The port forwarding thing was quite a bugger to figure out since the error is happening inside the detox app on the device. Perhaps it would be useful to have some sort of graphical feedback (notification, error message, etc) when the app faces an issue.

And for the `'Dozing'` state, I'm note entirely sure if this makes sense cross-device, I only tested on a Nokia 6.1 with Android 9.0, for which this was necessary.

File: detox/src/devices/drivers/AndroidDriver.js [hunk 1]
```diff
@@ -199,11 +200,16 @@ class AndroidDriver extends DeviceDriverBase {
...
     const additionalLaunchArgs = this._prepareLaunchArgs({debug: false});
+    const serverPort = new URL(this.client.configuration.server).port;
+    this.adb.reverse(deviceId, serverPort)
     const testRunner = await this.adb.getInstrumentationRunner(deviceId, bundleId);
     const spawnFlags = [`-s`, `${deviceId}`, `shell`, `am`, `instrument`, `-w`, `-r`, ...launchArgs, ...additionalLaunchArgs, testRunner];
 
     this.instrumentationProcess = spawnAndLog(this.adb.adbBin, spawnFlags, { detached: false });
-    this.instrumentationProcess.childProcess.on('close', () => this._terminateInstrumentation());
+    this.instrumentationProcess.childProcess.on('close', () => {
+      this._terminateInstrumentation()
```
Comment (noomorph) [inline detox/src/devices/drivers/AndroidDriver.js]: Could you rewrite to that to `async`-`await` style, please?
async-await-style

**Comment by noomorph** [comment]
Thank you for the pull request - it looks quite useful. I have one stylistic comment about async-await, and let's recheck what's up with CI build.

**Comment by mnzaki** [comment]
I rewrote with async/await.

The test failure is still mysterious to me though, it fails to install the app for one of the tests :/

**Comment by mnzaki** [comment]
Oh it actually passed. I think it was just probably an unrelated failure

**Comment by noomorph** [comment]
@mnzaki, thank you.

**Pull request merged**